keras_cv/models/segmentation/segformer/segformer.py

```diff
@@ -0,0 +1,86 @@
+from keras_cv.backend import keras
+from keras_cv.models import utils
+from keras_cv.models.task import Task
+
+
+@keras.utils.register_keras_serializable(package="keras_cv")
+class SegFormer(Task):
+    def __init__(
+        self,
+        num_classes=None,
+        backbone=None,
+        embed_dim=None,
+        input_shape=None,
+        input_tensor=None,
+        **kwargs,
+    ):
+        inputs = utils.parse_model_inputs(input_shape, input_tensor)
+        x = inputs
+        y = backbone(x)
```

Comment:
Because the backbone is expected to be MiT, we don't want pyramid-level outputs but instead the standard model output, correct?

If so, we should definitely mention in the docstring for SegFormer than only MiT/ViT are valid backbones

Reply:
We want the pyramid-level outputs here, which is the reason they modified a ViT-style arch to have hierarchical outputs.
Technically, this should work with a SwinTransformer-style output as well, or _technically_ any output that matches this signature:

```
{'P1': <KerasTensor shape=(None, 56, 56, 32), dtype=float32, name=keras_tensor_16>,
 'P2': <KerasTensor shape=(None, 28, 28, 64), dtype=float32, name=keras_tensor_31>,
 'P3': <KerasTensor shape=(None, 14, 14, 160), dtype=float32, name=keras_tensor_46>,
 'P4': <KerasTensor shape=(None, 7, 7, 256), dtype=float32, name=keras_tensor_61>}
```

This can arguably be any CNN. However, it's meant to be used with MiTs as the backbone.

I'll update this to operate on them instead of the final output (this was a temp line until MiTs are fully done, which can now be replaced with the correct outputs)